upload/src/addons/SV/CustomFieldPerms/XF/Entity/UserProfile.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace SV\CustomFieldPerms\XF\Entity;
+
+use XF\CustomField\Set;
+
+class UserProfile extends XFCP_UserProfile
+{
+
+	/**
+	 * Insert a new filter type into the DefinitionSet.
+	 * @return Set
+	 */
+	public function getCustomFields()
+	{
+		$set = parent::getCustomFields();
+
+		$set->getDefinitionSet()->addFilter('check_usergroup_perms', function(array $field, $usergroups, $keyWithPerms)
+		{
+			if ($field[ $keyWithPerms.'_enable' ]) {

```

Comment:
Should this have an `isset()` (or `!empty()`) check on the field? Especially to avoid wonky behaviour during upgrade deployments

Reply:
I guess, to be safe